### PR TITLE
Fix ROCm compilation

### PR DESCRIPTION
**Pull request by ariostas**
@slava77 found that in #381 there was an issue with the makefile causing `sdl_rocm` to fail to compile. I also found that `sdl_make_tracklooper` was not checking if this file was generated, which is probably why I missed this issue when working on #381.

### CODE REVIEW

**Comment by slava77** [comment]
is the TrackLooper-actions `standalone/run.sh` up to date to check ROCm builds?

**Comment by ariostas** [comment]
> is the TrackLooper-actions standalone/run.sh up to date to check ROCm builds?

Not yet, I'll do that once this PR gets merged.

File: Makefile [hunk 1]
```diff
@@ -21,10 +21,10 @@ ROOTCFLAGS  = $(foreach option, $(shell root-config --cflags), $(option))
 ALPAKAINCLUDE = -I${ALPAKA_ROOT}/include -I/${BOOST_ROOT}/include -std=c++17 -DALPAKA_DEBUG=0
 ALPAKA_CPU = -DALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED
 ALPAKA_CUDA = -DALPAKA_ACC_GPU_CUDA_ENABLED -DALPAKA_HOST_ONLY
-ALPAKA_ROCM = -DALPAKA_ACC_GPU_HIP_ENABLED -DALPAKA_HOST_ONLY -DALPAKA_DISABLE_VENDOR_RNG
-CFLAGS      = $(ROOTCFLAGS)  -Wall  -Wno-unused-function  -g  -O2  -fPIC  -fno-var-tracking -ISDL -I$(shell pwd) -Icode  -Icode/core -I${CUDA_HOME}/include  -fopenmp
+ALPAKA_ROCM = -DALPAKA_ACC_GPU_HIP_ENABLED -DALPAKA_HOST_ONLY -DALPAKA_DISABLE_VENDOR_RNG -D__HIP_PLATFORM_HCC__ -D__HIP_PLATFORM_AMD__
+CFLAGS      = $(ROOTCFLAGS)  -Wall  -Wno-unused-function  -g  -O2  -fPIC  -fno-var-tracking -ISDL -I$(shell pwd) -Icode  -Icode/core -I${CUDA_HOME}/include -I${ROCM_ROOT}/include -fopenmp
 EXTRACFLAGS = $(shell rooutil-config) -g
-EXTRAFLAGS  = -fPIC -ITMultiDrawTreePlayer -Wunused-variable -lTMVA -lEG -lGenVector -lXMLIO -lMLP -lTreePlayer -L${CUDA_HOME}/lib64 -lcudart -fopenmp
+EXTRAFLAGS  = -fPIC -ITMultiDrawTreePlayer -Wunused-variable -lTMVA -lEG -lGenVector -lXMLIO -lMLP -lTreePlayer -L${CUDA_HOME}/lib64 -lcudart -L${ROCM_ROOT}/lib -lamdhip64 -fopenmp
```
Comment (slava77) [inline Makefile:27]: would it make sense to factor out the flags by target? (cuda doesn't need to know about rocm; and cpu doesn't need either)

Comment (ariostas) [reply Makefile:27]: Yeah, that's true. That makefile already has multiple issues (#384) so I'll work on it.

**Review by slava77 — APPROVED** [review]

**Comment by ariostas** [comment]
The ROCm compilation is now being tested by the CI after https://github.com/SegmentLinking/TrackLooper-actions/pull/12. I tested that it works well in https://github.com/SegmentLinkingTests/TrackLooper/pull/19

**Pull request merged**